### ISSUE TITLE
About car number recognition using yolov5s

### ISSUE THREAD

**Issue by leeyunhome** [ultralytics/yolov5#3497]
## ❔Question

Hello,

Thank you for your always good help.
We are using yolov5s to recognize car numbers called lpr and anpr.
I am deducing from the jetson nano by labeling the number inside the license plate rather than recognizing only the license plate.

The accuracy is very poor.
We know that car license plate and number recognition is usually done by object detection up to license plate and numbering using ocr (optical character recognition), but we do not use ocr based on the excellent performance of yolov5s and all objects (license plate, license plate) genus number) by labeling and learning and inferring.

Is this a limitation of the jetson nano?
Models over yolov5s are not usable in jetson nano because they are slow.
Is it possible to recognize a car number with more than 95% accuracy using yolov5s?

If you know of a project that uses yolov5s to perform license plate recognition and number recognition, can you please let me know?

Thank you.

**Comment by glenn-jocher** [comment]
@leeyunhome I can't comment on your custom dataset, but in general for best results with small objects you'll want to run inference at large image sizes. A full guide on improving training results is below, including commentary on image sizes.

👋 Hello! Thanks for asking about improving training results. Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥1.5k images per class
- **Instances per class.** ≥10k instances (labeled objects) per class total
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total).

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. VOC, VisDrone, GlobalWheat). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. COCO, Objects365, OIv6). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyp.scratch.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains:
http://karpathy.github.io/2019/04/25/recipe/

**Comment by leeyunhome** [comment]
Hello,

Images per class. ≥**1.5k** images per class
**Instances** per class. ≥10k instances (labeled objects) per class total

1. What is the unit of 1.5k? Are you talking about 1500 images?
2. I don't know what the instance refers to exactly. Can you explain with an example that there is one image?
3. I have 42000 car license plates and number images that are labeled. Could this be enough?

Thank you.

**Comment by leeyunhome** [comment]
> Hello,
> 
> Images per class. ≥**1.5k** images per class
> **Instances** per class. ≥10k instances (labeled objects) per class total
> 
> 1. What is the unit of 1.5k? Are you talking about 1500 images?
> 2. I don't know what the instance refers to exactly. Can you explain with an example that there is one image?
> 3. I have 42000 car license plates and number images that are labeled. Could this be enough?

> Thank you.

**Comment by glenn-jocher** [comment]
@leeyunhome 1.5k indicates 1500 using the metric prefix k for kilo:
https://en.wikipedia.org/wiki/Metric_prefix

**Comment by GGDRriedel** [comment]
> 
> 
> Hello,
> 
> Images per class. ≥**1.5k** images per class
> **Instances** per class. ≥10k instances (labeled objects) per class total
> 
>     1. What is the unit of 1.5k? Are you talking about 1500 images?
> 
>     2. I don't know what the instance refers to exactly. Can you explain with an example that there is one image?
> 
>     3. I have 42000 car license plates and number images that are labeled. Could this be enough?
> 
> 
> Thank you.

When you have 1000 images with 2 license plates labeled "license plate" per image you have 2k instances on 1k images

If you have 42000 license plates labeled, if we are talking just the numbers 0 to 9 you have ~ on average ~ 4200 instances per class if a license plate would just contain one number. 

Here in Germany we have numbers and letters, so 10+26 classes =36 classes for the signs. 
If one license plate had just one sign you'd have on average 1.166 instances per class. 

Best case: your 42000 is not license plates but license plates with labeled signs, so lets say a license plate has 6 signs a-b, 0-9 that would make 42000*6 labels or 42000*6/36 7000 instances per class

If however your 42000 labeled images are "just" labeled letters/numbers you'd have 42000/36 = 1166 instances per class
